### PR TITLE
Addition of Trivial Acceptance Tests

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description
Addition of trivial acceptance tests

### CODE REVIEW

**Comment by bastelfreak** [comment]
We need a new modulesync to update https://github.com/voxpupuli/puppet-systemd/blob/master/.github/workflows/ci.yml. when the tests are present and modulesync is run, the CI file is updated to run the acceptance tests.

**Review by bastelfreak — APPROVED** [review]

File: spec/acceptance/systemd_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+require 'spec_helper_acceptance'
+
+describe 'systemd' do
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      include systemd
+      PUPPET
+    end
```
Comment (ekohl) [inline spec/acceptance/systemd_spec.rb]: Only partly related discussion and you probably know this, but the short version of this is:
```suggestion
    let(:manifest) { 'include systemd' }
```

I wonder if we should encourage one style over the other or just see what individuals prefer.

**Review by ekohl — COMMENTED** [review]
I like this. Acceptance tests are so useful.

**Comment by traylenator** [comment]
This is what are feared, you very quickly start running to weird things about these hosts:

```
Error: /Stage[main]/Systemd::Resolved/File[/etc/resolv.conf]/ensure: change from 'file' to 'link' failed: Device or resource busy @ apply2files - /etc/resolv.conf
```

File: spec/acceptance/systemd_spec.rb (hunk 1 above)
Comment (traylenator) [reply spec/acceptance/systemd_spec.rb]: There HERE document has the advantage that you can correct puppet style quoting once the line becomes more interesting.

Comment (ekohl) [reply spec/acceptance/systemd_spec.rb]: I agree and I think for consistency that's why I also commonly use heredoc. It makes it easy to add multiple lines without reformatting the whole block.

File: spec/acceptance/resolved_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    it 'works idempotently with no errors' do
+      pp = <<-PUPPET
+      class{'systemd':
+        manage_resolved    => true,
+        manage_resolv_conf => false,
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 requires an extra run
+      apply_manifest(pp, catch_failures: true)
```
Comment (bastelfreak) [inline spec/acceptance/resolved_spec.rb:16]: can you wrap that in a condition so it only runs twice on RHEL9? Also why are two runs required? that shouldn't be the case at all?

**Comment by traylenator** [comment]
That;s probably enough in here for now:

* I'll pull out the `manage_resolv_conf` to it's own MR.
* Disable the checks on CentOS as there is indeed no resolved to be had.

File: spec/acceptance/resolved_spec.rb (hunk 2 above)
Comment (ekohl) [reply spec/acceptance/resolved_spec.rb:16]: Perhaps rephrase it to "extra run to populate $facts['internal_services']"?

**Comment by traylenator** [comment]
#256 first for the extra parameter.

File: spec/acceptance/resolved_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        manage_resolv_conf => #{default[:hypervisor] != 'docker'},
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 and newer installs package first run before fact  $facts['internal_services'] is set
+      apply_manifest(pp, catch_failures: true) if !%w[7 8].include?(fact('os.release.major')) && (fact('os.family') == 'RedHat')
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    # RedHat 7 does not have systemd-resolved available at all.
```
Comment (ekohl) [inline spec/acceptance/resolved_spec.rb:20]: On CentOS 7 there is a `systemd-resolved` package with a `/usr/lib/systemd/system/systemd-resolved.service`. In practice I noticed it was harder to use due to a lack of `resolvectl` so I ended up not using it, but it's not entirely correct. I'd be surprised if RHEL 7 and CentOS 7 differed in this.

Comment (traylenator) [reply spec/acceptance/resolved_spec.rb:20]: Indeed that package  is available CentOS 7 also, just not in 8. :-(

File: spec/acceptance/resolved_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    it 'works idempotently with no errors' do
+      pp = <<-PUPPET
+      class{'systemd':
+        manage_resolved    => true,
+        manage_resolv_conf => #{default[:hypervisor] != 'docker'},
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 and newer installs package first run before fact  $facts['internal_services'] is set
+      apply_manifest(pp, catch_failures: true) if Gem::Version.new(fact('os.release.major')) >= Gem::Version.new('9') && (fact('os.family') == 'RedHat')
```
Comment (ekohl) [inline spec/acceptance/resolved_spec.rb:16]: This would be more explicit. Not insisting on it, but it could help the reader?
```suggestion
      separate_resolved_package = Gem::Version.new(fact('os.release.major')) >= Gem::Version.new('9') && (fact('os.family') == 'RedHat')
      apply_manifest(pp, catch_failures: true) if separate_resolved_package
```

File: spec/acceptance/resolved_spec.rb (hunk 3 above)
Comment (ekohl) [reply spec/acceptance/resolved_spec.rb:20]: Does that mean we should include the package name in the Hiera data?

File: spec/acceptance/resolved_spec.rb (hunk 4 above)
Comment (traylenator) [reply spec/acceptance/resolved_spec.rb:16]: With the comment I think I prefer, seems more obvious what to delete when C8 is dead.
